### PR TITLE
Adds support for ManyCam

### PR DESCRIPTION
**Pull request by kingbuzzman**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by kingbuzzman** [comment]
I'm having issues installing/uninstalling the app. My ruby is not that great, i'd appreciate greatly some help.

*I get this when installing it.*
```
==> Verifying SHA-256 checksum for Cask 'manycam'.
==> Installing Cask manycam
==> Running installer for manycam; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are ignored.
Password:
Error: SIGTERM
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/system_command.rb:157:in `select'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:157:in `block in each_line_from'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:156:in `loop'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:156:in `each_line_from'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:143:in `each_output_line'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:47:in `run!'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:35:in `run'
/usr/local/Homebrew/Library/Homebrew/system_command.rb:39:in `run!'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/pkg.rb:60:in `block in run_installer'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/pkg.rb:66:in `with_choices_file'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/pkg.rb:53:in `run_installer'
/usr/local/Homebrew/Library/Homebrew/cask/artifact/pkg.rb:35:in `install_phase'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:219:in `block in install_artifacts'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:777:in `each'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/set.rb:777:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:212:in `install_artifacts'
/usr/local/Homebrew/Library/Homebrew/cask/installer.rb:104:in `install'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:73:in `block in install_casks'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:72:in `each'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:72:in `install_casks'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/install.rb:37:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd/abstract_command.rb:95:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:186:in `run'
/usr/local/Homebrew/Library/Homebrew/cask/cmd.rb:120:in `run'
/usr/local/Homebrew/Library/Homebrew/cmd/cask.rb:14:in `cask'
/usr/local/Homebrew/Library/Homebrew/brew.rb:119:in `<main>'
```
*Uninstalling it says all is good, but nothing gets deleted.*
```
==> Uninstalling Cask manycam
==> Purging files for version 7.5.0.7 of Cask manycam
```

**Comment by miccal** [comment]
@reitermarkus and @vitorgalvao I have tested this locally, and although the install phase outputs `Error: SIGTERM` the install in fact _does_ succeed, and the `uninstall` and `zap` stanzas both work as intended.

I am not sure about the `Error: SIGTERM` fault, though -- any thoughts?

The only other alternative I can think of is an `installer manual`.

Also note that I removed the `uninstall script` as it calls a GUI.

**Comment by vitorgalvao** [comment]
> I am not sure about the `Error: SIGTERM` fault, though -- any thoughts?

It’s coming from the the PKG’s `preinstall` script, lines 56 to 61 (specifically line 59):

```bash
echo "Terminate suspect processes..."
for ((i=0; i<$mc_pids_count; i++))
do
sudo kill -15 ${mc_pids[$i]}
echo "kill -15 ${mc_pids[$i]} = $?"
done
```

In case you’re not familiar with those, PKGs can have (and most do) a `preinstall` and `postinstall` scripts to run arbitrary actions. They’re easy to inspect if you unpack the PKG (`brew install vitorgalvao/tiny-scripts/pkg-extract` is a script to do just that).

> have tested this locally, and (…) the install in fact _does_ succeed, and the `uninstall` and `zap` stanzas both work as intended.

Then it should be merged. A working local install shouldn’t be compromised due to a failing CI (unless the cask constantly has that problem), and ideally the CI should be fixed. But if feel free to add `ci-skip-install` (that’s what it’s for) and continue as you see best.

**Comment by miccal** [comment]
@vitorgalvao I inpected the `preinstall` and `postinstall` scripts inside the `pkg` file, thanks. 

How do you know that it is specifically the line `sudo kill -15 ${mc_pids[$i]}` that raises this error?

And is there a way we can suppress this error when this Cask is installed?

File: Casks/manycam.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  uninstall pkgutil:   "com.visicom.ManyCam.installer.pkg",
+            quit:      "com.visicom.ManyCam.application",
+            launchctl: "com.visicom.ManyCam.VideoDevice.agent"
+
+  zap trash: [
+    "~/Library/Application Support/Visicom Media",
+    "~/Library/Preferences/com.visicom-media.ManyCam.plist",
+    "~/Library/Saved Application State/com.visicom.ManyCam.application.savedState",
+    "~/Pictures/ManyCam",
```
Comment (kingbuzzman) [inline Casks/manycam.rb:24]: I would argue that this _should_ stay.. no? This is strictly user data.

Comment (kingbuzzman) [reply Casks/manycam.rb]: Thank you @vitorgalvao

**Comment by vitorgalvao** [comment]
> How do you know that it is specifically the line `sudo kill -15 ${mc_pids[$i]}` that raises this error?

`kill -15` is `SIGTERM` (the `-15` is redundant, seeing as it’s the default). I should’ve been clear I don’t *know*  that’s the cause because I haven’t explored (tested) that far, but it seems like a likely culprit.

> And is there a way we can suppress this error when this Cask is installed?

Not that I’m aware of. Maybe @reitermarkus can confirm.

**Comment by kingbuzzman** [comment]
Any update on this?

**Comment by miccal** [comment]
> Any update on this?

I have tried many different ways to suppress this error, but to no avail.

I have requested a review from our Cask guru, and I am sure that they will provide assistance when they are able -- as you can imagine, many maintainers are quite busy with the release of Big Sur.

**Comment by kingbuzzman** [comment]
Yes, terrible timing on my part, thank you for the update @miccal .

**Comment by vitorgalvao** [comment]
> I have tested this locally, and although the install phase outputs `Error: SIGTERM` the install in fact _does_ succeed, and the `uninstall` and `zap` stanzas both work as intended.

Does that pose an issue when installing the cask locally, or only for CI?

**Comment by miccal** [comment]
> Does that pose an issue when installing the cask locally, or only for CI?

The install succeeds locally, but is not tracked by Cask -- that is, it would not show as installed via `brew list --casks`.

**Comment by vitorgalvao** [comment]
> The install succeeds locally, but is not tracked by Cask -- that is, it would not show as installed via `brew list --casks`.

That is indeed a problem. And it does seem to be our bug, because I tried to install it with the `/usr/sbin/installer` command we use and it returned `0`.

**Comment by kingbuzzman** [comment]
@vitorgalvao please let me know if there is any new information regarding this bug. Thanks

**Comment by vitorgalvao** [comment]
I have no clue why this happens. If we can’t figure it out, we’ll unfortunately have to reject the cask.

**Comment by kingbuzzman** [comment]
@vitorgalvao should i just delete this PR?

**Comment by vitorgalvao** [comment]
If the issue still happens then we should close, yes. Perhaps we’ll be able to track it down another time. Thank you.

**Comment by kingbuzzman** [comment]
Sounds good, closing it for now.

**Pull request closed without merging**